Non blocking SSL does not use read or write properly.  With TLS 1.2
it works by accident.  Do not use TLS 1.3.

Index: lib/Net/Server/Proto/SSLEAY.pm
--- lib/Net/Server/Proto/SSLEAY.pm.orig
+++ lib/Net/Server/Proto/SSLEAY.pm
@@ -150,6 +150,7 @@ sub bind_SSL {
     my $ctx = Net::SSLeay::CTX_new();  $sock->SSLeay_check_fatal("SSLeay bind_SSL CTX_new");
 
     Net::SSLeay::CTX_set_options($ctx, Net::SSLeay::OP_ALL());  $sock->SSLeay_check_fatal("SSLeay bind_SSL CTX_set_options");
+    Net::SSLeay::CTX_set_max_proto_version($ctx, Net::SSLeay::TLS1_2_VERSION());
 
     # 0x1:  SSL_MODE_ENABLE_PARTIAL_WRITE
     # 0x10: SSL_MODE_RELEASE_BUFFERS (ignored before OpenSSL v1.0.0)
